### PR TITLE
[youtube] Fix extraction of age restricted videos

### LINKED ISSUE #666: NullPointerException on specific Video
I could replicate this on both NewPipe and Piped.
YouTube says: "The uploader has not made this video available in your country."

The video is: https://www.youtube.com/watch?v=DYCEXg4_lCM

In such a case, it would be nice to throw a ContentNotAvailable exception.

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=DYCEXg4_lCM
* __Content Country:__ US
* __Content Language:__ en
* __App Language:__ en
* __Service:__ YouTube
* __Version:__ 0.21.5
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
	at java.io.StringReader.<init>(StringReader.java:50)
	at com.grack.nanojson.JsonParser$JsonParserContext.from(Unknown Source:6)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.fetchVideoInfoPage(YoutubeStreamExtractor.java:802)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:731)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:54)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:68)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:115)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$YTHJjScxCJNO1LTCqs3IKy35iyY.call(Unknown Source:4)
	at io.reactivex.rxjava3.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:43)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:15753)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.rxjava3.core.Maybe.subscribe(Maybe.java:5330)
	at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
	at io.reactivex.rxjava3.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.rxjava3.core.Scheduler$DisposeTask.run(Scheduler.java:614)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
	at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</details>
<hr>

### PR DESCRIPTION
**Pull request by B0pol**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

`sts` is no longer required so we no longer request `/embed/:id` to get `sts`

See https://github.com/ytdl-org/youtube-dl/issues/29333

Closes #666, closes TeamNewPipe/NewPipe#6403

### CODE REVIEW

File: extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderFactory.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
     public final static String RESOURCE_PATH = "src/test/resources/org/schabi/newpipe/extractor/";
 
-    private final static DownloaderType DEFAULT_DOWNLOADER = DownloaderType.REAL;
```
Comment (TobiGr) [inline extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderFactory.java:11]: I guess you did not want to add this line

**Comment by TiA4f8R** [comment]
The `sts` param is required to get cipher protected URLs: if some age-restricted URLs have signature ciphers, streams will return 403. So you should keep the `sts` param (I fixed its extraction in #604, don't fix it in this PR).

**Comment by B0pol** [comment]
> The `sts` param is required to get cipher protected URLs: if some age-restricted URLs have signature ciphers, streams will return 403. So you should keep the `sts` param (I fixed its extraction in #604, don't fix it in this PR).

but sts is empty

**Comment by TobiGr** [comment]
Does the `sts` extraction work for some videos? If that is the case, we should re-add the line. Otherwise, that can be done in #604

**Comment by B0pol** [comment]
No because the regex changed.

https://github.com/TeamNewPipe/NewPipeExtractor/blob/35c1bdd0120fafe1c572d1f31826aab05f943f6e/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java#L834

fixed:

https://github.com/TeamNewPipe/NewPipeExtractor/blob/5f4f123b4c907136eef1af5f6de396422293294e/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java#L1033

https://github.com/TeamNewPipe/NewPipeExtractor/blob/5f4f123b4c907136eef1af5f6de396422293294e/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java#L693

**Review by TiA4f8R — CHANGES_REQUESTED** [review]
Just a small thing and this PR can be merged.

**Review by TiA4f8R — APPROVED** [review]

**Pull request merged**